### DISCUSSION TITLE
TensorFlow pipeline filter on FluentBit - HowTo

### DISCUSSION THREAD

**Discussion by canob** [fluent/fluent-bit#8135] [Q&A]
Hi everyone,
Is there any additional information or blog post of how to use TensorFlow pipeline filter on FluentBit that you know?
Because I reviewed the docs, https://docs.fluentbit.io/manual/pipeline/filters/tensorflow, and is not clear for me or there are not many examples.
I'm trying to do "Anomaly Detection" on the edge. I'm reviewing another options, like OpenSearch Data Prepper Anomaly Detector processor, https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/anomaly-detector/, but my best option is only use FluentBit for all the pipeline processing, including Anomaly Detection.
Thanks in advance for your help!

**Comment by patrick-stephens** [comment]
You might be better asking in the Slack channel: https://slack.fluentd.org/
I would also say please add any examples or docs afterwards if you can to help future users: https://github.com/fluent/fluent-bit-docs/blob/master/pipeline/filters/tensorflow.md

There's also Observability Day at KubeCon this week to chat to maintainers directly.

**Comment by canob** [comment]
Hi @patrick-stephens , thanks for your answer.
I already asked that on Slack many days ago, but no answer, and that was the reason to ask here.
If I find a way to make it work, obviously I'm going to give an example here to help other users, but for now, I didn't.

**Comment by koleini** [comment]
Hi @canob, I am the author of the TensorFlow plugin for Fluent Bit. You can find examples in this repository (the external plugin support GPU inference as well):

https://github.com/koleini/fluent-bit-external-plugins/tree/main/filter_tensorflow

You can also directly message me on Fluent Bit slack channel if you had questions or facing issues.

**Comment by canob** [comment]
Thanks @koleini ! I'm going to review the provided link to try to understand how to use the plugin.

**Comment by koleini** [comment]
If you decided to use NVIDIA Jetson Nano board, this external Fluent Bit plugin can read images from the camera and pass them to the TensorFlow plugin for inference:

https://github.com/koleini/fluent-bit-external-plugins/tree/main/in_csi_camera
